airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java

```diff
@@ -214,27 +211,55 @@ private static Container getMain(final String image,
     return containerBuilder.build();
   }
 
-  private static void copyFilesToKubeConfigVolume(final ApiClient officialClient,
-                                                  final String podName,
-                                                  final String namespace,
+  private static void copyFilesToKubeConfigVolume(final KubernetesClient client,
+                                                  final Pod podDefinition,
                                                   final Map<String, String> files) {
     final List<Map.Entry<String, String>> fileEntries = new ArrayList<>(files.entrySet());
 
     // copy this file last to indicate that the copy has completed
     fileEntries.add(new AbstractMap.SimpleEntry<>(SUCCESS_FILE_NAME, ""));
 
+    Path tmpFile = null;
+    Process proc = null;
     for (final Map.Entry<String, String> file : fileEntries) {
       try {
+        tmpFile = Path.of(IOs.writeFileToRandomTmpDir(file.getKey(), file.getValue()));
+
         LOGGER.info("Uploading file: " + file.getKey());
-        final var contents = file.getValue().getBytes(StandardCharsets.UTF_8);
         final var containerPath = Path.of(CONFIG_DIR + "/" + file.getKey());
 
-        // fabric8 kube client upload doesn't work on gke:
-        // https://github.com/fabric8io/kubernetes-client/issues/2217
-        final Copy copy = new Copy(officialClient);
-        copy.copyFileToPod(namespace, podName, INIT_CONTAINER_NAME, contents, containerPath);
-      } catch (final IOException | ApiException e) {
+        // using kubectl cp directly here, because both fabric and the official kube client APIs have
+        // several issues with copying files. See https://github.com/airbytehq/airbyte/issues/8643 for
+        // details.
+        final String command = String.format("kubectl cp %s %s/%s:%s -c %s", tmpFile, podDefinition.getMetadata().getNamespace(),
+            podDefinition.getMetadata().getName(), containerPath, INIT_CONTAINER_NAME);
+        LOGGER.info(command);
+
+        proc = Runtime.getRuntime().exec(command);
+        LOGGER.info("Waiting for kubectl cp to complete");
+        final int exitCode = proc.waitFor();
+
+        if (exitCode != 0) {
+          // Copying the success indicator file to the init container causes the container to immediately
+          // exit, causing the `kubectl cp` command to exit with code 137. This check ensures that an error is
+          // not thrown in this case if the init container exits successfully.
+          if (file.getKey().equals(SUCCESS_FILE_NAME) && waitForInitPodToTerminate(client, podDefinition, 10, TimeUnit.SECONDS) == 0) {
```

Comment:
Do we need to check the status? What if the init pod terminated with an error?

Reply:
That's what the `== 0` is checking; I wrote the `waitForInitPodToTerminate` method to return the exit value of the init container